walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,52 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.Collection; 
+import java.util.Set;
+import java.util.HashSet;
+import java.util.*;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> available_times = new ArrayList<TimeRange>();
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) { // not possible for duration to exceed whole day
+        return available_times;
+    }
+    ArrayList<TimeRange> blocked_times = new ArrayList<TimeRange>();
+    Set<String> r_attendees = new HashSet<String>(request.getAttendees());      
+    for (Event e : events) { // check each event of the day for blocked times
+        Set<String> e_attendees = new HashSet<String>(e.getAttendees());
+        Set<String> intersection = new HashSet<String>(r_attendees);
+        intersection.retainAll(e_attendees);
+        if (intersection.isEmpty()) { 
+            continue; // none of our attendees will be at this event
+        }
+        else {
+            blocked_times.add(e.getWhen()); // at least one of our attendees will be at this event so we cannot do this time
+        }
+    }
+    Collections.sort(blocked_times, TimeRange.ORDER_BY_START);
+    
+    // remove any nested events
+    for (int count = 0; count < blocked_times.size()-1; count++) {
+        if ((blocked_times.get(count)).contains(blocked_times.get(count+1))) {
+            blocked_times.remove(count+1);
+            count--;
```

Comment:
I think this means you can get to `count == -1` on the first iteration and so get out of bounds on the second.

Reply:
If you get to count == -1 on the first iteration and go back to the top of the for loop won't it become count == 0 after the for loop updates count again (doing count++) ? So it will stay in bounds?

Reply:
The `count++` statement is executed at the end of the for loop. 

If you get to count == -1 in the first iteration and go back to the top of the for loop, count will still be -1 until it goes through the loop one more time. Then `blockedTimes.get(-1)` will throw an out of bounds exception.